lib/scheduler.pm

```diff
@@ -86,6 +86,24 @@ sub get_test_suite_data {
     return $test_suite_data;
 }
 
+=head2 expand_test_data_vars
+
+Expand test suite data variables
+
+=cut
+
+sub expand_test_data_vars {
+    my ($node) = shift;
+    if (ref $node eq 'HASH') {
+        foreach my $k (keys %$node) {
+            $node->{$k} =~ s/%(.*?)%/get_var($1)/eg if !ref $node->{$k};
```

Comment:
What's about the case if variable is not defined? Did you check what openQA does in such case? In anycase we should not fail here with attempt to replace line with undef. I would propose either set empty string as default value for `get_var`, or not to define it, depending on what openQA does to be consistent.

Reply:
I checked, it didn't fail, for example it printed: "SLES-" without expansion.

Reply:
Do you have logs, because I've written small code snipper:
```perl
sub mmm {
    return undef;
}

my $ttt = "%AAA%";
$ttt =~ s/%(.*?)%/mmm()/eg;
```
And I get `Use of uninitialized value in substitution iterator`. Documentation and the code of os-autoinst will return `undef`. I guess you mean that program didn't crash, but I'm quite confident that you had this exception in the output somewhere, like we have for concat of undef sometimes.

Reply:
And yeah, following code:
```perl
sub mmm {
    return undef;
}

my $ttt = "bzz%AAA%";
$ttt =~ s/%(.*?)%/mmm()/eg;

print($ttt);
```
Prints `bzz`, but still with the error being thrown.

Reply:
I can see it in the logs, good catch. let me check.